### ISSUE TITLE
Unable to figure out how to do simple a upload :-)

### ISSUE THREAD

**Issue by nyh** [leocrawford/picasawebsync#27]
Until recently, I was using GoogleCL (https://code.google.com/p/googlecl/) to upload my photo albums to Picasaweb. Sadly, last month this stops working as Google phased out OAuth1 support, and nobody is working on upgrading GoogleCL to OAuth2. So I was happy to find out that Picasawebsync uses OAuth2, does still work.

However, I've been unable to get it to do the simplest upload I want. What I want is to take a directory of photos, and upload it to an album with a name that I choose. I also want the ability to add photos to an existing album (bonus points if it "syncs" automatically figuring out which new files should be added to the album).

My first problem with picasawebsync is that when it starts, it always "scans" all the albums in my Picasa. I have hundreds of these albums, spanning tens of thousands of photos, so this process takes ages and constantly keeps complaining about duplicate images, I'm not even sure why. But all these old albums are not relevant to the new pictures I'm trying to upload! To solve this problem, I'm using "--skipserver", like this:

```
picasawebsync.py --skipserver '*' -v -d ~/pictures/2015/05/eilat
```

One consequence of this is that this command always creates a new album - even if the album with the name it chooses already exists in picasaweb. I don't know if I was meant to do something else...

My second problem is that in the above example, picasawebsync insists to call the output folder "Home" (because of the "/home" in the path). I read the documentation about all sorts of clever album naming, but I don't really want it - I just want to choose a specific name myself. Can I do that?

My third problem is that picasawebsync insists of adding a caption to each photo - it's file name. This is really ugly, and I really don't want that. How can I disable it?

Thanks,
Nadav.

**Comment by leocrawford** [comment]
> However, I've been unable to get it to do the simplest upload I want. 
> What I want is to take a directory of photos, and upload it to an 
> album with a name that I choose. I also want the ability to add 
> photos to an existing album (bonus points if it "syncs" automatically 
> figuring out which new files should be added to the album).
> 
> My first problem with picasawebsync is that when it starts, it always 
> "scans" all the albums in my Picasa. I have hundreds of these albums, 
> spanning tens of thousands of photos, so this process takes ages and 
> constantly keeps complaining about duplicate images, I'm not even 
> sure why. But all these old albums are not relevant to the new 
> pictures I'm trying to upload! To solve this problem, I'm using 
> "--skipserver", like this:
> 
> picasawebsync.py --skipserver '*' -v -d ~/pictures/2015/05/eilat
> One consequence of this is that this command always creates a new 
> album - even if the album with the name it chooses already exists in 
> picasaweb. I don't know if I was meant to do something else...
> 

The program is design to sync the whole repository, not a folder so 
it's not unexpected this doesn't work.

As a quick work around you could create a glob that matches every 
directory except your one, e.g. if it is called sync

[!s][!y][!n][!c]

I'm assuming ! works in python glob expressions, but haven't checked.

If you would like to submit a patch to handle directory syncing 
properly I'd be keen to accept it.

> My second problem is that in the above example, picasawebsync insists 
> to call the output folder "Home" (because of the "/home" in the 
> path). I read the documentation about all sorts of clever album 
> naming, but I don't really want it - I just want to choose a specific 
> name myself. Can I do that?
> 
Yes and no.

The script runs relative to the directory specified using -d, so simply 
set that to be /home/sync (or whatever directory you're trying to sync)
> My third problem is that picasawebsync insists of adding a caption to 
> each photo - it's file name. This is really ugly, and I really don't 
> want that. How can I disable it?
> 

You may find it ugly, but it's needed if you want to do a restore or 
sync (which is its purpose). It's not hard to disable, it's just one of 
code, so feel free to clone and modify if you wish.

Leo

**Comment by nyh** [comment]
Thanks for the answer. I guess this is not a bug, per se (so I'm closing this issue), but a very different use case that picasawebsync handles: It seems picasawebsync is wonderful for the case where somebody uses it to create and maintain a new photo collection, but it cannot be used without modification to add new albums to a huge pre-existing collection (as I said, I already have tens of thousands of photos in hundreds of albums in my Picasa web).

Anyway, I modified the code to add an option to sync only one specific remote folder (instead of excluding a regexp), and it worked well. My patch is ugly, so I don't want to post it, but it's fairly trivial.

About the file-name captions: there's a way to write the file name without making them a caption (or "title" in picasa terms), which is very visible on the web interface. If I look at photos I uploaded with the now defunct googlecl, they show a "Filename:" when I click on "Photo information", so I guess it's stored somewhere. Moreover, in the old days (up to about a year ago), I was able to use Google's actual Picasa software to sync photos, and their sync also worked without these captions (I have no idea if it used filenames stored not in captions, or perhaps more likely - they could have use checksums instead of filenames, as those are more reliable).

Anyway, thanks for writing this code - it's the only *working* picasa uploader I could find, and I can modify it to do exactly what I want, so it's great. Thanks!
